internal/serve/httphandler/forgot_password_handler_test.go

```diff
@@ -57,51 +61,79 @@ func Test_ForgotPasswordHandler(t *testing.T) {
 		SDPUIBaseURL: &uiBaseURL,
 	}
 	ctx := tenant.SaveTenantInContext(context.Background(), &tnt)
+	const email = "valid@email.com"
+	user := &auth.User{
+		ID:    "userID",
+		Email: email,
+		Roles: []string{data.OwnerUserRole.String()},
+	}
+	const reCAPTCHAToken = "validToken"
+	defaultReqBody := fmt.Sprintf(`
+				{ 
+					"email": "%s",
+					"recaptcha_token": "%s"
+				}`, user.Email, reCAPTCHAToken)
 
 	t.Run("Should return http status 200 on a valid request", func(t *testing.T) {
-		requestBody := `
-		{ 
-			"email": "valid@email.com" ,
-			"recaptcha_token": "validToken"
-		}`
-
-		rr := httptest.NewRecorder()
-		req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, strings.NewReader(requestBody))
-		require.NoError(t, err)
-
-		authenticatorMock.
-			On("ForgotPassword", req.Context(), "valid@email.com").
-			Return("resetToken", nil).
-			Once()
-		reCAPTCHAValidatorMock.
-			On("IsTokenValid", mock.Anything, "validToken").
-			Return(true, nil).
-			Once()
-
-		resetPasswordLink, err := urllib.JoinPath(uiBaseURL, "reset-password")
-		require.NoError(t, err)
-
-		content, err := htmltemplate.ExecuteHTMLTemplateForForgotPasswordMessage(htmltemplate.ForgotPasswordMessageTemplate{
-			ResetToken:        "resetToken",
-			ResetPasswordLink: resetPasswordLink,
-			OrganizationName:  "MyCustomAid",
-		})
-		require.NoError(t, err)
-
-		msg := message.Message{
-			ToEmail: "valid@email.com",
-			Title:   forgotPasswordMessageTitle,
-			Message: content,
+		usersRoles := [][]string{
+			{data.OwnerUserRole.String()},                            // API roles cannot bypass reCAPTCHA
+			{data.OwnerUserRole.String(), data.APIUserRole.String()}, // API roles can bypass reCAPTCHA
 		}
-		messengerClientMock.
-			On("SendMessage", msg).
-			Return(nil).
-			Once()
+		for _, userRoles := range usersRoles {
+			targetUser := &auth.User{
+				ID:    "user-ID",
```

Comment:
A few lines above you used `userID` and here you have `user-ID`. Is that a typo? Is that field being used for anything? Should we add a test for it?